<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Vladyslav Husak">

<title>network-based analysis project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Final project - Vladyslav Husak_files/libs/clipboard/clipboard.min.js"></script>
<script src="Final project - Vladyslav Husak_files/libs/quarto-html/quarto.js"></script>
<script src="Final project - Vladyslav Husak_files/libs/quarto-html/popper.min.js"></script>
<script src="Final project - Vladyslav Husak_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Final project - Vladyslav Husak_files/libs/quarto-html/anchor.min.js"></script>
<link href="Final project - Vladyslav Husak_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Final project - Vladyslav Husak_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Final project - Vladyslav Husak_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Final project - Vladyslav Husak_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Final project - Vladyslav Husak_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">network-based analysis project</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Vladyslav Husak </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="network-based-analysis-project" class="level2">
<h2 class="anchored" data-anchor-id="network-based-analysis-project">Network-based analysis project</h2>
<p>Author: Vladyslav Husak</p>
</section>
<section id="data-loading-normalization-and-prepossessing" class="level2">
<h2 class="anchored" data-anchor-id="data-loading-normalization-and-prepossessing">Data loading, normalization and prepossessing</h2>
<p>The data is loaded from recount3 database. Then raw counts are scaled (<code>transform_counts(my_rse)</code>) and then saved to .csv files (both raw and normalized).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'NProject.RData'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"recount3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"recount"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>projects <span class="ot">&lt;-</span> <span class="fu">available_projects</span>()</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">"SRP101294"</span>, projects<span class="sc">$</span>project))</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>proj_info <span class="ot">&lt;-</span> <span class="fu">subset</span>(</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  projects,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  project <span class="sc">==</span> <span class="st">"SRP101294"</span> <span class="sc">&amp;</span> project_type <span class="sc">==</span> <span class="st">"data_sources"</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>my_rse <span class="ot">&lt;-</span> <span class="fu">create_rse</span>(proj_info)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>cols_meta <span class="ot">&lt;-</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(my_rse))</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(my_rse, <span class="st">"counts"</span>) <span class="ot">&lt;-</span> <span class="fu">transform_counts</span>(my_rse)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>counts_all <span class="ot">&lt;-</span> <span class="fu">assay</span>(my_rse, <span class="st">"counts"</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(counts_all, <span class="st">'counts_norm.csv'</span>, <span class="at">sep =</span> <span class="st">','</span>, <span class="at">quote =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">assay</span>(my_rse, <span class="st">"raw_counts"</span>), <span class="st">'counts_raw.csv'</span>, <span class="at">sep =</span> <span class="st">','</span>, <span class="at">quote =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Creating metadata table</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>description <span class="ot">&lt;-</span> my_rse<span class="sc">$</span>sra.sample_attributes</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>sam_id <span class="ot">&lt;-</span> my_rse<span class="sc">$</span>external_id</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>sam_titles <span class="ot">&lt;-</span> my_rse<span class="sc">$</span>sra.sample_title</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>meta_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sam_id  =</span> sam_id , <span class="at">sam_title =</span> sam_titles, <span class="at">description =</span> description)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> description, </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"gender"</span>, <span class="st">"source_name"</span>, <span class="st">"time"</span>, <span class="st">"tissue"</span>), </span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>,</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">remove =</span> <span class="cn">TRUE</span>, </span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">convert =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">age =</span> <span class="fu">as.integer</span>(<span class="fu">gsub</span>(<span class="st">"[age;]"</span>, <span class="st">""</span>, age)))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gender =</span> (<span class="fu">gsub</span>(<span class="st">"gender;;"</span>, <span class="st">""</span>, gender)))</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">source_name =</span> (<span class="fu">gsub</span>(<span class="st">"source_name;;"</span>, <span class="st">""</span>, source_name)))</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">time =</span> (<span class="fu">gsub</span>(<span class="st">"time;;"</span>, <span class="st">""</span>, time)))</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tissue =</span> (<span class="fu">gsub</span>(<span class="st">"tissue;;"</span>, <span class="st">""</span>, tissue)))</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(df_meta, <span class="st">'metadata.csv'</span>, <span class="at">sep =</span> <span class="st">','</span>, <span class="at">quote =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Loading data from saved files (to omit reloading from the database)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"metadata.csv"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>counts_all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"counts_norm.csv"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>counts_raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"counts_raw.csv"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>df_norm_counts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(counts_all))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>df_raw_counts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(counts_raw))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="influential-genes-selection-with-classification-models" class="level2">
<h2 class="anchored" data-anchor-id="influential-genes-selection-with-classification-models">Influential genes selection with classification models</h2>
<p>Firstly, to reduce the number of parameters to be feasible for models with minimal loss of important variance, the variance threshold was set.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>variance_threshold <span class="ot">&lt;-</span> <span class="dv">50000</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate variance for each column</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>variances <span class="ot">&lt;-</span> <span class="fu">apply</span>(df_norm_counts, <span class="dv">2</span>, var)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter columns to remove those with variance below the threshold</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>df_reduced_var <span class="ot">&lt;-</span> df_norm_counts[, variances <span class="sc">&gt;</span> variance_threshold]</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(df_reduced_var)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>df_reduced_var<span class="sc">$</span>Group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(df_meta<span class="sc">$</span>time)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(df_reduced_var<span class="sc">$</span>Group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'caret'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    lift</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Matrix'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    expand, pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loaded glmnet 4.1-8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: foreach</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'foreach'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    accumulate, when</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: iterators</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: parallel</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">detectCores</span>() <span class="sc">-</span> <span class="dv">2</span>) <span class="co"># Two cores for the OS</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">registerDoParallel</span>(cl)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)  <span class="co"># For reproducibility</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>train_control <span class="ot">&lt;-</span> <span class="fu">trainControl</span>(</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"cv"</span>,  <span class="co"># Cross-validation</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">number =</span> <span class="dv">10</span>,    <span class="co"># Number of folds</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">savePredictions =</span> <span class="st">"final"</span>,</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">classProbs =</span> <span class="cn">TRUE</span>,</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">allowParallel =</span> TRUE<span class="co"># If you need probability scores for ROC, etc.</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(df_reduced_var<span class="sc">$</span>Group) <span class="ot">&lt;-</span> <span class="fu">make.names</span>(<span class="fu">levels</span>(df_reduced_var<span class="sc">$</span>Group), <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>lda_model <span class="ot">&lt;-</span> <span class="fu">train</span>(Group <span class="sc">~</span> ., <span class="at">data =</span> df_reduced_var, <span class="at">method =</span> <span class="st">"lda"</span>,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trControl =</span> train_control)</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>elas_model <span class="ot">&lt;-</span> <span class="fu">train</span>(Group <span class="sc">~</span> ., <span class="at">data =</span> df_reduced_var, <span class="at">method =</span> <span class="st">"glmnet"</span>,</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>                     <span class="at">trControl =</span> train_control,</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tuneLength =</span> <span class="dv">10</span>,  <span class="co"># Search over 10 values of lambda</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>                     <span class="at">family =</span> <span class="st">"binomial"</span>)  <span class="co"># Assuming binary classification</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> <span class="fu">train</span>(Group <span class="sc">~</span> ., <span class="at">data =</span> df_reduced_var, <span class="at">method =</span> <span class="st">"rf"</span>,</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trControl =</span> train_control,</span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ntree =</span> <span class="dv">100</span>)  <span class="co"># Number of trees</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">resamples</span>(<span class="fu">list</span>(<span class="at">LDA =</span> lda_model, <span class="at">RF =</span> rf_model,</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Elastic_Net =</span> elas_model))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(results, <span class="at">main =</span> <span class="st">'Models</span><span class="sc">\'</span><span class="st"> evaluation on classification performance'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>elas_res <span class="ot">&lt;-</span> elas_model<span class="sc">$</span>results</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>ordered_results <span class="ot">&lt;-</span> elas_res[<span class="fu">order</span>(<span class="sc">-</span>elas_res<span class="sc">$</span>Accuracy),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>gr <span class="ot">&lt;-</span> ordered_results <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(alpha) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Accuracy =</span> <span class="fu">mean</span>(Accuracy))</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">y=</span>gr<span class="sc">$</span>Accuracy, <span class="at">x=</span>gr<span class="sc">$</span>alpha, <span class="at">main =</span> <span class="st">'Elastic Net grid search'</span>, <span class="at">xlab =</span> <span class="st">'Alpha'</span>, <span class="at">ylab =</span> <span class="st">'Accuracy'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Training the model of the whole data set to select the most influential parameters</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_reduced_var, <span class="at">select =</span> <span class="sc">-</span>Group)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>trainIndex <span class="ot">&lt;-</span> <span class="fu">createDataPartition</span>(df_reduced_var<span class="sc">$</span>Group, <span class="at">p =</span> .<span class="dv">7</span>, <span class="at">list =</span> <span class="cn">FALSE</span>, <span class="at">times =</span> <span class="dv">1</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create training and testing sets</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>train_data <span class="ot">&lt;-</span> df_reduced_var[trainIndex, ]</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>test_data <span class="ot">&lt;-</span> df_reduced_var[<span class="sc">-</span>trainIndex, ]</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>x_train <span class="ot">&lt;-</span> <span class="fu">subset</span>(train_data, <span class="at">select =</span> <span class="sc">-</span>Group)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>x_test <span class="ot">&lt;-</span> <span class="fu">subset</span>(test_data, <span class="at">select =</span> <span class="sc">-</span>Group)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>cv_elas <span class="ot">&lt;-</span> <span class="fu">cv.glmnet</span>(<span class="fu">sapply</span>(x_train, as.double), train_data<span class="sc">$</span>Group, <span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">family=</span><span class="st">'binomial'</span>)</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>best_lambda <span class="ot">&lt;-</span> cv_elas<span class="sc">$</span>lambda.min</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>validation_model <span class="ot">&lt;-</span> <span class="fu">glmnet</span>(<span class="fu">sapply</span>(x_train, as.double), train_data<span class="sc">$</span>Group, </span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">family =</span> <span class="st">'binomial'</span>, <span class="at">lambda =</span> best_lambda)</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(validation_model, <span class="at">newx =</span> <span class="fu">sapply</span>(x_test, as.double), <span class="at">type =</span> <span class="st">"class"</span>)</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>actual_outcomes <span class="ot">&lt;-</span> test_data<span class="sc">$</span>Group</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate accuracy</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>accuracy <span class="ot">&lt;-</span> <span class="fu">mean</span>(predictions <span class="sc">==</span> actual_outcomes)</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(accuracy)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>final_model <span class="ot">&lt;-</span> <span class="fu">glmnet</span>(<span class="fu">sapply</span>(x, as.double), df_reduced_var<span class="sc">$</span>Group, </span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">family =</span> <span class="st">'binomial'</span>, <span class="at">lambda =</span> best_lambda)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vip)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>importance_data <span class="ot">&lt;-</span> <span class="fu">vi</span>(final_model, <span class="at">lambda =</span> best_lambda)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>top_500_features <span class="ot">&lt;-</span> <span class="fu">head</span>(importance_data[<span class="fu">order</span>(<span class="sc">-</span>importance_data<span class="sc">$</span>Importance), ], <span class="dv">500</span>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>top_500_features<span class="sc">$</span>base_ids <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, top_500_features<span class="sc">$</span>Variable)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>gene_names500 <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'ensembl_gene_id'</span>, <span class="st">'external_gene_name'</span>),</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">filters =</span> <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values =</span> top_500_features<span class="sc">$</span>base_ids,</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mart =</span> ensembl)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame from the fetched results</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>gene_names_df500 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ensembl_gene_id =</span> gene_names500<span class="sc">$</span>ensembl_gene_id,</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>                            <span class="at">gene_name =</span> gene_names500<span class="sc">$</span>external_gene_name,</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>                            <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge the fetched gene names with the top_500_features dataframe</span></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>top_500_features <span class="ot">&lt;-</span> <span class="fu">merge</span>(top_500_features, gene_names_df500, <span class="at">by.x =</span> <span class="st">"base_ids"</span>, <span class="at">by.y =</span> <span class="st">"ensembl_gene_id"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>top_500_features_clean <span class="ot">&lt;-</span> top_500_features[<span class="sc">!</span>(top_500_features<span class="sc">$</span>gene_name <span class="sc">%in%</span> <span class="st">""</span>),]</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>top_features <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(top_500_features_clean)</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_raw_counts) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(df_raw_counts))</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_reduced_var) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(df_reduced_var))</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>top_genes_ids <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(top_features<span class="sc">$</span>base_ids)</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>final_count_matrix500_raw <span class="ot">&lt;-</span> df_raw_counts[, <span class="fu">colnames</span>(df_raw_counts) <span class="sc">%in%</span> top_genes_ids]</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>final_count_matrix500 <span class="ot">&lt;-</span> df_reduced_var[, <span class="fu">colnames</span>(df_reduced_var) <span class="sc">%in%</span> top_genes_ids]</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>mapping_df_ordered500_raw <span class="ot">&lt;-</span> top_features[<span class="fu">match</span>(<span class="fu">colnames</span>(final_count_matrix500_raw), top_features<span class="sc">$</span>base_ids), ]</span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>mapping_df_ordered500 <span class="ot">&lt;-</span> top_features[<span class="fu">match</span>(<span class="fu">colnames</span>(final_count_matrix500), top_features<span class="sc">$</span>base_ids), ]</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(final_count_matrix500_raw) <span class="ot">&lt;-</span> mapping_df_ordered500_raw<span class="sc">$</span>gene_name</span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(final_count_matrix500) <span class="ot">&lt;-</span> mapping_df_ordered500<span class="sc">$</span>gene_name</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-expression-and-gene-enrichment" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-and-gene-enrichment">Differential expression and gene enrichment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># preparing metadate for DE analysis</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"sam_id"</span>, <span class="st">"time"</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>meta_des <span class="ot">&lt;-</span> df_meta[, keep, drop<span class="ot">=</span>F]</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(meta_des)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"condition"</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>meta_des<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(meta_des<span class="sc">$</span>condition)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(meta_des<span class="sc">$</span>condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="using-scaled-counts" class="level3">
<h3 class="anchored" data-anchor-id="using-scaled-counts">Using scaled counts</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Matrix':

    expand, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    second, second&lt;-</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:lubridate':

    %within%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grDevices':

    windows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biobase'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:MatrixGenerics':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>dds500 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">t</span>(final_count_matrix500),</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> meta_des,</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>dds500 <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds500)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 8 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>res500 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds500)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>res_df500 <span class="ot">&lt;-</span> <span class="fu">select</span>(<span class="fu">data.frame</span>(res500), <span class="fu">c</span>(<span class="st">'log2FoldChange'</span>, <span class="st">'pvalue'</span>))</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>res_df500 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">Gene.symbol =</span> <span class="fu">rownames</span>(res_df500), res_df500)</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res_df500)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"GENE"</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res_df500)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"CHANGE"</span></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(res_df500) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pathfindR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: pathfindR.data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
                        Welcome to pathfindR!

Please cite the article below if you use pathfindR in published reseach:

Ulgen E, Ozisik O, Sezerman OU. 2019. pathfindR: An R Package for Comprehensive
Identification of Enriched Pathways in Omics Data Through Active Subnetworks.
Front. Genet. doi:10.3389/fgene.2019.00858

##############################################################################</code></pre>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>RA <span class="ot">&lt;-</span> <span class="fu">run_pathfindR</span>(res_df500, <span class="at">iterations=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>## Testing input</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The input looks OK</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Processing input. Converting gene symbols,
          if necessary (and if human gene symbols provided)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Number of genes provided in input: 480</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Number of genes in input after p-value filtering: 335</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pathfindR cannot handle p values &lt; 1e-13. These were changed to 1e-13</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Could not find any interactions for 28 (8.36%) genes in the PIN</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Final number of genes in input: 307</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Performing Active Subnetwork Search and Enrichment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 79 active subnetworks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Processing the enrichment results over all iterations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Annotating involved genes and visualizing enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Plotting the enrichment bubble chart</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 35 enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>You may run:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>- cluster_enriched_terms() for clustering enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>- visualize_terms() for visualizing enriched term diagrams</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>RA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         ID                                Term_Description Fold_Enrichment
1  hsa03083                     Polycomb repressive complex        4.093739
2  hsa01212                           Fatty acid metabolism       11.789967
3  hsa05167 Kaposi sarcoma-associated herpesvirus infection        2.497874
4  hsa05169                    Epstein-Barr virus infection        2.069868
5  hsa05163                 Human cytomegalovirus infection        2.962303
6  hsa05220                        Chronic myeloid leukemia        4.093739
7  hsa01040         Biosynthesis of unsaturated fatty acids       13.397690
8  hsa04668                           TNF signaling pathway        2.188732
9  hsa03040                                     Spliceosome        4.641719
10 hsa05142                                  Chagas disease        4.557977
11 hsa03082              ATP-dependent chromatin remodeling        2.600728
12 hsa04625        C-type lectin receptor signaling pathway        2.377010
13 hsa04350                      TGF-beta signaling pathway        2.210619
14 hsa00100                            Steroid biosynthesis        3.878279
15 hsa00030                       Pentose phosphate pathway        9.824973
16 hsa04936                         Alcoholic liver disease        3.792728
17 hsa04150                          mTOR signaling pathway        3.203795
18 hsa01200                               Carbon metabolism        4.131998
19 hsa05203                            Viral carcinogenesis        2.472728
20 hsa05164                                     Influenza A        2.408082
21 hsa05202         Transcriptional misregulation in cancer        1.435467
22 hsa05170        Human immunodeficiency virus 1 infection        2.497874
23 hsa03320                          PPAR signaling pathway        6.140608
24 hsa04218                             Cellular senescence        2.005097
25 hsa05161                                     Hepatitis B        1.483637
26 hsa04744                               Phototransduction        2.631689
27 hsa04975                    Fat digestion and absorption        4.093739
28 hsa04060          Cytokine-cytokine receptor interaction        1.625455
29 hsa05417                       Lipid and atherosclerosis        1.151364
30 hsa04130       SNARE interactions in vesicular transport        2.302728
31 hsa04110                                      Cell cycle        1.524565
32 hsa05020                                   Prion disease        3.524175
33 hsa05171                  Coronavirus disease - COVID-19        2.344596
34 hsa04144                                     Endocytosis        1.000280
35 hsa05322                    Systemic lupus erythematosus        5.024134
   occurrence    support     lowest_p    highest_p
1           1 0.05063291 7.327128e-08 7.327128e-08
2           1 0.02531646 3.820658e-06 3.820658e-06
3           1 0.01265823 5.668906e-04 5.668906e-04
4           1 0.03797468 5.797578e-04 5.797578e-04
5           1 0.01265823 9.035460e-04 9.035460e-04
6           1 0.03797468 1.212026e-03 1.212026e-03
7           1 0.02531646 2.061517e-03 2.061517e-03
8           1 0.01265823 2.362442e-03 2.362442e-03
9           1 0.02531646 2.398329e-03 2.398329e-03
10          1 0.01265823 2.978964e-03 2.978964e-03
11          1 0.02531646 4.730734e-03 4.730734e-03
12          1 0.01265823 4.782594e-03 4.782594e-03
13          1 0.02531646 5.947083e-03 5.947083e-03
14          1 0.01265823 6.194035e-03 6.194035e-03
15          1 0.01265823 7.461477e-03 7.461477e-03
16          1 0.01265823 8.209857e-03 8.209857e-03
17          1 0.01265823 8.576192e-03 8.576192e-03
18          1 0.01265823 1.003836e-02 1.003836e-02
19          1 0.03797468 1.078473e-02 1.078473e-02
20          1 0.02531646 1.167217e-02 1.167217e-02
21          1 0.02531646 1.190129e-02 1.190129e-02
22          1 0.02531646 1.268650e-02 1.268650e-02
23          1 0.01265823 1.705658e-02 1.705658e-02
24          1 0.01265823 1.880951e-02 1.880951e-02
25          1 0.01265823 1.958145e-02 1.958145e-02
26          1 0.01265823 2.193647e-02 2.193647e-02
27          1 0.01265823 2.275160e-02 2.275160e-02
28          1 0.01265823 2.281819e-02 2.281819e-02
29          1 0.01265823 2.295000e-02 2.295000e-02
30          1 0.01265823 2.302613e-02 2.302613e-02
31          1 0.01265823 2.336493e-02 2.336493e-02
32          1 0.01265823 2.765133e-02 2.765133e-02
33          1 0.01265823 3.437033e-02 3.437033e-02
34          1 0.01265823 3.483520e-02 3.483520e-02
35          1 0.01265823 4.380814e-02 4.380814e-02
                                          Up_regulated
1                                   PHC2, SCMH1, HDAC1
2                                                 PPT1
3                               NFKBIA, PIK3CD, NFATC4
4                        HDAC1, NFKBIA, PIK3CD, SAP30L
5                       PIK3CD, NFKBIA, NFATC4, PRKACB
6                                HDAC1, NFKBIA, PIK3CD
7                                                     
8                             NFKBIA, TNFRSF1B, PIK3CD
9  SF3A3, PCBP1, SRSF10, SRSF4, SRSF6, SYF2, PPIE, FUS
10              C1QA, C1QB, C1QC, NFKBIA, GNAL, PIK3CD
11                                        HDAC1, MEAF6
12                              NFKBIA, NFATC4, PIK3CD
13                                         NBL1, HDAC1
14                                                    
15                                                H6PD
16                     NFKBIA, C1QA, C1QB, C1QC, ADH1C
17                              PIK3CD, RPS6KA1, EIF4B
18                                                H6PD
19                       PIK3CD, HDAC1, NFKBIA, PRKACB
20                              TPSAB1, NFKBIA, PIK3CD
21                                          FUS, HDAC1
22                    PIK3CD, NFATC4, NFKBIA, TNFRSF1B
23                                               FABP3
24                                      PIK3CD, NFATC4
25                              NFKBIA, NFATC4, PIK3CD
26                                                    
27                                      PLA2G2A, ABCA1
28            TNFRSF14, CSF3R, TNFRSF1B, TNFSF12, LEPR
29                               PIK3CD, NFKBIA, ABCA1
30                                                    
31                                               HDAC1
32                    PRKACB, C1QA, C1QB, C1QC, PIK3CD
33       NFKBIA, RPS8, RPL11, PIK3CD, C1QA, C1QB, C1QC
34                                        ASAP3, SMAP2
35                                    C1QA, C1QB, C1QC
                                    Down_regulated
1                                            RBBP4
2  MECR, ELOVL1, SCD, FADS2, HSD17B4, ACADL, ACADM
3                                CDK6, GNB1, GNG12
4                                             CDK6
5                        GNB1, GNG12, PTGER3, CDK6
6                                             CDK6
7                      FADS2, HSD17B4, ELOVL1, SCD
8                                                 
9                                                 
10                                                
11                                           RBBP4
12                                                
13                                             SKI
14                                          DHCR24
15                                ALDOC, PGD, PGM1
16                                      SCD, ACADM
17                             RRAGC, RNF152, FZD4
18                   GAPDH, ALDOC, PGD, ENO1, SDHB
19                                            CDK6
20                                     KPNA6, CDK6
21                                          CDKN2C
22                                     GNB1, GNG12
23                   ACADM, ACADL, SCD, FADS2, DBI
24                                     CDK6, RBBP4
25                                                
26                                            GNB1
27                                                
28                                        TNFRSF25
29                                                
30                                           VAMP3
31                                    CDK6, CDKN2C
32          SDHB, UQCRHL, UQCRH, COX2, PSMB2, CAV1
33                                                
34                                            CAV1
35                                                </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>RA_clu <span class="ot">&lt;-</span> <span class="fu">cluster_enriched_terms</span>(RA)</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a><span class="fu">term_gene_graph</span>(RA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df500[<span class="fu">order</span>(res_df500<span class="sc">$</span>pvalue),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        GENE     CHANGE       pvalue
308    FADS2 -1.7159695 9.204478e-72
351    ALDOC -1.2152604 6.486193e-67
438 CCDC162P -1.2474906 8.493006e-59
428    SPARC -0.6373703 2.855707e-51
439   ECHDC1 -0.6744896 2.957906e-50
317    EIF4B  0.3199847 2.469002e-48</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df500[<span class="fu">order</span>(res_df500<span class="sc">$</span>CHANGE),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        GENE    CHANGE       pvalue
304      SCD -1.856826 2.712761e-47
350    SCDP1 -1.836355 2.789448e-41
308    FADS2 -1.715970 9.204478e-72
438 CCDC162P -1.247491 8.493006e-59
351    ALDOC -1.215260 6.486193e-67
474    MXRA5 -1.081321 9.340524e-29</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df500[<span class="fu">order</span>(res_df500<span class="sc">$</span>CHANGE, <span class="at">decreasing =</span> T),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         GENE    CHANGE       pvalue
319       DCD 3.0639635 9.438642e-13
427      GPX3 0.7536289 3.736827e-43
30     GPR153 0.6217646 1.125409e-24
368      APOE 0.6211502 1.004725e-11
355 RAMP2-AS1 0.6087268 1.394525e-32
385     CPXM1 0.5915592 2.464076e-13</code></pre>
</div>
</div>
</section>
<section id="using-raw-counts" class="level3">
<h3 class="anchored" data-anchor-id="using-raw-counts">Using raw counts</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>dds500_raw <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">t</span>(final_count_matrix500_raw),</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> meta_des,</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>dds500_raw <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds500_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 8 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>res500_raw <span class="ot">&lt;-</span> <span class="fu">results</span>(dds500_raw)</span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>res_df500_raw <span class="ot">&lt;-</span> <span class="fu">select</span>(<span class="fu">data.frame</span>(res500_raw), <span class="fu">c</span>(<span class="st">'log2FoldChange'</span>, <span class="st">'pvalue'</span>))</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>res_df500_raw <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">Gene.symbol =</span> <span class="fu">rownames</span>(res_df500_raw), res_df500_raw)</span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res_df500_raw)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"GENE"</span></span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res_df500_raw)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"CHANGE"</span></span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(res_df500_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a>RA_raw <span class="ot">&lt;-</span> <span class="fu">run_pathfindR</span>(res_df500_raw, <span class="at">iterations=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>## Testing input</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The input looks OK</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Processing input. Converting gene symbols,
          if necessary (and if human gene symbols provided)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Number of genes provided in input: 480</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Number of genes in input after p-value filtering: 334</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pathfindR cannot handle p values &lt; 1e-13. These were changed to 1e-13</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Could not find any interactions for 28 (8.38%) genes in the PIN</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Final number of genes in input: 306</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Performing Active Subnetwork Search and Enrichment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 68 active subnetworks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Processing the enrichment results over all iterations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Annotating involved genes and visualizing enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Plotting the enrichment bubble chart</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 32 enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>You may run:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>- cluster_enriched_terms() for clustering enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>- visualize_terms() for visualizing enriched term diagrams</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>RA_clu_raw <span class="ot">&lt;-</span> <span class="fu">cluster_enriched_terms</span>(RA_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>The maximum average silhouette width was 0.3 for k = 13 </code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-24-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">term_gene_graph</span>(RA_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-24-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df500_raw[<span class="fu">order</span>(res_df500_raw<span class="sc">$</span>pvalue),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        GENE     CHANGE       pvalue
308    FADS2 -1.7160747 1.003169e-71
351    ALDOC -1.2153046 7.053458e-67
438 CCDC162P -1.2475536 9.226944e-59
428    SPARC -0.6374071 3.043733e-51
439   ECHDC1 -0.6745314 3.054992e-50
317    EIF4B  0.3199681 2.578386e-48</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df500_raw[<span class="fu">order</span>(res_df500_raw<span class="sc">$</span>CHANGE, <span class="at">decreasing =</span> T),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         GENE    CHANGE       pvalue
319       DCD 3.7033121 4.046817e-15
427      GPX3 0.7535956 3.945965e-43
30     GPR153 0.6217496 1.212529e-24
368      APOE 0.6210867 1.020696e-11
355 RAMP2-AS1 0.6085845 1.592681e-32
385     CPXM1 0.5915036 2.558479e-13</code></pre>
</div>
</div>
</section>
</section>
<section id="network-building-without-features-selection" class="level2">
<h2 class="anchored" data-anchor-id="network-building-without-features-selection">Network-building without features selection</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>ddsall <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">t</span>(df_raw_counts),</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> meta_des,</span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): 45 duplicate rownames
were renamed by adding numbers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a>ddsall <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsall)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 15447 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a>resall <span class="ot">&lt;-</span> <span class="fu">results</span>(ddsall)</span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a>res_dfall <span class="ot">&lt;-</span> <span class="fu">select</span>(<span class="fu">data.frame</span>(resall), <span class="fu">c</span>(<span class="st">'log2FoldChange'</span>, <span class="st">'pvalue'</span>))</span>
<span id="cb174-3"><a href="#cb174-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-4"><a href="#cb174-4" aria-hidden="true" tabindex="-1"></a>res_dfall <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">Gene.symbol =</span> <span class="fu">rownames</span>(res_dfall), res_dfall)</span>
<span id="cb174-5"><a href="#cb174-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res_dfall)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"GENE"</span></span>
<span id="cb174-6"><a href="#cb174-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res_dfall)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"CHANGE"</span></span>
<span id="cb174-7"><a href="#cb174-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(res_dfall) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_dfall[<span class="fu">order</span>(res_dfall<span class="sc">$</span>pvalue),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 GENE    CHANGE       pvalue
28462 ENSG00000265168 -1.302232 4.365230e-67
28463 ENSG00000109107 -1.239089 1.311248e-66
52510 ENSG00000203799 -1.267323 1.425432e-59
8751  ENSG00000163207 27.373440 1.820075e-58
14777 ENSG00000173727 -1.029373 1.967190e-56
14559 ENSG00000134824 -1.743618 2.630215e-54</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a>gene_names_all <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'ensembl_gene_id'</span>, <span class="st">'external_gene_name'</span>),</span>
<span id="cb177-2"><a href="#cb177-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">filters =</span> <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb177-3"><a href="#cb177-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values =</span> res_dfall<span class="sc">$</span>GENE,</span>
<span id="cb177-4"><a href="#cb177-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mart =</span> ensembl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>resall_names <span class="ot">=</span> <span class="fu">merge</span>(res_dfall, gene_names_all, <span class="at">by.x =</span> <span class="st">"GENE"</span>, <span class="at">by.y =</span> <span class="st">"ensembl_gene_id"</span>, <span class="at">all.x =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resall_names[<span class="fu">order</span>(resall_names<span class="sc">$</span>pvalue),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 GENE    CHANGE       pvalue external_gene_name
49991 ENSG00000265168 -1.302232 4.365230e-67                   
3639  ENSG00000109107 -1.239089 1.311248e-66              ALDOC
19673 ENSG00000203799 -1.267323 1.425432e-59           CCDC162P
10952 ENSG00000163207 27.373440 1.820075e-58                IVL
13587 ENSG00000173727 -1.029373 1.967190e-56              FAUP4
6970  ENSG00000134824 -1.743618 2.630215e-54              FADS2</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a>resall_names<span class="sc">$</span>GENE <span class="ot">=</span> resall_names<span class="sc">$</span>external_gene_name</span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a>resall_names<span class="sc">$</span>external_gene_name <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb181-3"><a href="#cb181-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pathfindR)</span>
<span id="cb181-4"><a href="#cb181-4" aria-hidden="true" tabindex="-1"></a>RA_all <span class="ot">&lt;-</span> <span class="fu">run_pathfindR</span>(<span class="fu">na.omit</span>(resall_names), <span class="at">iterations=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>## Testing input</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The input looks OK</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Processing input. Converting gene symbols,
          if necessary (and if human gene symbols provided)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Number of genes provided in input: 54486</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Number of genes in input after p-value filtering: 12410</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Duplicated genes found! The lowest p value for each gene was selected</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pathfindR cannot handle p values &lt; 1e-13. These were changed to 1e-13</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Could not find any interactions for 2228 (21.03%) genes in the PIN</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Final number of genes in input: 8366</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Performing Active Subnetwork Search and Enrichment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 185 active subnetworks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Processing the enrichment results over all iterations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>## Annotating involved genes and visualizing enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Plotting the enrichment bubble chart</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 255 enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>You may run:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>- cluster_enriched_terms() for clustering enriched terms</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>- visualize_terms() for visualizing enriched term diagrams</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final-project---Vladyslav-Husak_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resall_names)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    GENE      CHANGE       pvalue
1 TSPAN6 -0.05745950 4.293532e-02
2   TNMD -0.37270334 1.396387e-06
3   DPM1 -0.10893654 2.052784e-03
4  SCYL3 -0.03713178 5.221587e-02
5  FIRRM -0.03591146 9.145691e-02
6    FGR  0.39255138 4.497114e-04</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="at">file =</span> <span class="st">"NProject.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>